pip_audit/format/cyclonedx.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List
+
+from cyclonedx import output  # type: ignore
+from cyclonedx.model.bom import Bom  # type: ignore
+from cyclonedx.model.component import Component  # type: ignore
+from cyclonedx.model.vulnerability import Vulnerability  # type: ignore
+from cyclonedx.parser import BaseParser  # type: ignore
+
+import pip_audit.service as service
+
+from .interface import VulnerabilityFormat
+
+
+class PipAuditResultParser(BaseParser):
+    def __init__(self, result: Dict[service.Dependency, List[service.VulnerabilityResult]]):
+        super().__init__()
+
+        for (dep, vulns) in result.items():
+            c = Component(name=dep.name, version=str(dep.version))
+
+            # TODO(ww): Figure out if/how we want to include other dependency
+            # metadata in the BOM, such as author, license, etc.
+
+            for vuln in vulns:
+                # TODO(ww): Figure out if/how we want to include other vulnerability
+                # metadata in the BOM, such as source (OSV/PyPI/etc), URL, etc.
```

Comment:
CycloneDX is smart enough generate a PyPI "PURL" identifier for us, so maybe this isn't necessary. But if we do want more metadata in each `VulnerabilityResult`, we could do something similar to the `Metadata` I'm thinking about above.